### PR TITLE
located errors: database

### PR DESCRIPTION
**Pull request by da2ce7**
Related Issue: #176 

Depends #177 

This is the first in a series of work to add locations to error messages. I have created the [located_error.rs](https://github.com/da2ce7/torrust-tracker/blob/database-error-overhall/src/located_error.rs) class that gathers location from the error call, and make use of it for the database errors.

### CODE REVIEW

**Comment by da2ce7** [comment]
This pull request is Ready to Review.

File: tests/api/asserts.rs [hunk 1]
```diff
@@ -37,9 +37,20 @@ pub async fn assert_auth_key_utf8(response: Response) -> AuthKey {
...
+    let response_status = response.status().clone();
+    let response_headers = response.headers().get("content-type").cloned().unwrap();
+    let response_text = response.text().await.unwrap();
+
+    let details = format!(
+        r#"
+   status: ´{response_status}´
+  headers: ´{response_headers:?}´
+     text: ´"{response_text}"´"#
+    );
```
Comment (josecelano) [inline tests/api/asserts.rs:49]: hey @da2ce7 , this is an interesting way to have a kind of "object literal" or "anonymous struct". It's related to https://github.com/torrust/torrust-tracker/pull/172

I would not include it in this PR unless we change all the asserts to follow the same pattern. Maybe the final solution could be a mix of this solution and what I proposed on that [PR](https://github.com/torrust/torrust-tracker/pull/172). In this solution, I still miss the error pointing to the origin line when the condition was not fulfilled.

**Review by josecelano — APPROVED** [review]

**Comment by josecelano** [comment]
I think this is a very helpful improvement.

ack 9a3ba0668efd7543e45265518cb35e622d6e5a63

**Comment by da2ce7** [comment]
closing in favor of #178

**Pull request closed without merging**